e2e/utils/zetacore.go

```diff
@@ -448,3 +481,56 @@ func WaitForZetaBlocks(
 		}
 	}
 }
+
+// WaitForZRC20BalanceChange waits for the zrc20 balance of the given address to change by the given delta amount
+// This function is to tolerate the fact that the balance update may not be synced across all nodes behind a RPC.
+func WaitForZRC20BalanceChange(
+	t require.TestingT,
+	zrc20 *zrc20.ZRC20,
+	address common.Address,
+	oldBalance *big.Int,
+	change BalanceChange,
+	logger infoLogger,
+) {
+	// wait until the expected balance is reached or timeout
+	startTime := time.Now()
+	checkInterval := constant.ZetaBlockTime / 2
```

Comment:
Looks overenginereed to me
I don't think the RPC sync issue is linked to Zeta block time?
A simpler model to me look to jsut retry a few time every seconds

Reply:
Uses 2 secs static number instead: https://github.com/zeta-chain/node/pull/4053/commits/50e81ef434bd202d9e9b554984a063b6fbb433c9